config/m4/compiler.m4

```diff
@@ -195,6 +195,19 @@ AC_COMPILE_IFELSE([AC_LANG_SOURCE([[
                 CFLAGS="$SAVE_CFLAGS"])
 
 
+#
+# Set default datatype alignment to 16 bytes
```

Comment:
this would have effect only on clang, right?
do we want to add something to ucs_malloc() to make sure alignment is at least 16b?

Reply:
as I remember malloc guarantee 16 byte alignment

Reply:
you are right, but i would add some assertion in ucs_malloc and friends just in case, maybe even just for documentation purposes. maybe AC_DEFINE(UCX_ALLOC_ALIGN, 16) in here (compiler.m4), and use it in ucs/memtrack.h?

Reply:
ok, done

Reply:
@yosefe I bit refactored m4 scripts, https://github.com/openucx/ucx/pull/3401 is not relevant anymore